back/tests/Feature/Message/UpdateActivated/UpdateActivatedDeveloperTest.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Tests\Feature\Message\Update;
+
+use App\Models\Role;
+use Illuminate\Http\JsonResponse;
+use Tests\TestCaseWithAuth;
+
+class UpdateActivatedDeveloperTest extends TestCaseWithAuth
+{
+    protected $username = Role::DEVELOPER;
+
+    /**
+     * @group message
+     */
+    public function testDeveloperUpdatingActivatedMessage()
+    {
+        $message = $this->messageProvider();
+
+        $active = [
+            'is_active' => true,
+        ];
+
+        $response = $this->json('PATCH', route('messages.updateActivated', ['message_id' => $message->id]), $active)
+            ->assertStatus(JsonResponse::HTTP_CREATED)
+            ->assertJsonStructure([
+                'id',
+                'title',
+                'content',
+                'link',
+                'is_active',
+                'created_at',
+                'updated_at',
+                'deleted_at',
+            ]);
+
+
+        $content = json_decode($response->getContent(), true);
+        $this->assertTrue($content['is_active']);
```

Comment:
Il faudrait aussi que tu checkes que tous les autres messages dans la bdd sont bien désactivés dans le cas de l'activation d'un message. Pour celà n'hésite pas à créer 1 ou 2 autres messages actifs en utilisant `$this->messageActiveProvider()` et vérifie qu'ils sont bien désactivés quand tu lances ton test sur le 3e.

Aussi, je te conseille de faire un autre test pour tester la désactivation d'un message. Tu peux le faire que pour le test "Développeur" parce que c'est pareil pour "Administrateur" donc pas la peine de dupliquer du code.